sigstore/_cli.py

```diff
@@ -228,6 +238,102 @@ def _parser() -> argparse.ArgumentParser:
+        "attest",
+        help="sign one or more inputs using DSSE",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        parents=[parent_parser],
+    )
+    attest.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    dsse_options = attest.add_argument_group("DSSE options")
+    dsse_options.add_argument(
+        "--predicate",
+        metavar="FILE",
+        type=Path,
+        required=True,
+        help="Path to the predicate file",
+    )
+    dsse_options.add_argument(
+        "--predicate-type",
+        metavar="TYPE",
+        choices=PREDICATE_TYPES_CLI_MAP,
+        type=str,
+        required=True,
+        help=f"Specify a predicate type ({'|'.join(PREDICATE_TYPES_CLI_MAP)})",
+    )
+
+    oidc_options = attest.add_argument_group("OpenID Connect options")
+    oidc_options.add_argument(
+        "--identity-token",
+        metavar="TOKEN",
+        type=str,
+        default=os.getenv("SIGSTORE_IDENTITY_TOKEN"),
+        help="the OIDC identity token to use",
+    )
+    _add_shared_oidc_options(oidc_options)
+
+    output_options = attest.add_argument_group("Output options")
+    output_options.add_argument(
+        "--no-default-files",
+        action="store_true",
+        default=_boolify_env("SIGSTORE_NO_DEFAULT_FILES"),
+        help="Don't emit the default output files ({input}.sigstore.json)",
+    )
+    output_options.add_argument(
+        "--signature",
+        "--output-signature",
+        metavar="FILE",
+        type=Path,
+        default=os.getenv("SIGSTORE_OUTPUT_SIGNATURE"),
+        help=(
+            "Write a single signature to the given file; does not work with multiple input files"
+        ),
+    )
+    output_options.add_argument(
+        "--certificate",
+        "--output-certificate",
+        metavar="FILE",
+        type=Path,
+        default=os.getenv("SIGSTORE_OUTPUT_CERTIFICATE"),
+        help=(
+            "Write a single certificate to the given file; does not work with multiple input files"
+        ),
+    )
+    output_options.add_argument(
+        "--bundle",
+        metavar="FILE",
+        type=Path,
+        default=os.getenv("SIGSTORE_BUNDLE"),
+        help=(
+            "Write a single Sigstore bundle to the given file; does not work with multiple input "
+            "files"
+        ),
+    )
+    output_options.add_argument(
+        "--output-directory",
+        metavar="DIR",
+        type=Path,
+        default=os.getenv("SIGSTORE_OUTPUT_DIRECTORY"),
+        help=(
+            "Write default outputs to the given directory (conflicts with --signature, --certificate"
+            ", --bundle)"
+        ),
+    )
+    output_options.add_argument(
+        "--overwrite",
+        action="store_true",
+        default=_boolify_env("SIGSTORE_OVERWRITE"),
+        help="Overwrite preexisting signature and certificate outputs, if present",
+    )
```

Comment:
Given that this is a brand new command, I'm kind of tempted to remove/avoid the need to do anything except emit bundles 🙂 -- I _think_ the only options we need here are `--bundle` and `--overwrite`.

Reply:
The advantage of having these is that it allows us to reuse most of the code that was already written for `sigstore sign`. But I can remove them and refactor things around a bit.

Reply:
I think this still needs to be done -- IMO it's okay if end up being able to create a reusable `_sign_common`, since these extra flags are about the same size as the helper anyays (and we don't want to support them).

Reply:
Fixed! See last commit